lib/dbservice_web/controllers/session_schedule_controller.ex

```diff
@@ -0,0 +1,131 @@
+defmodule DbserviceWeb.SessionScheduleController do
+  use DbserviceWeb, :controller
+
+  import Ecto.Query
+  alias Dbservice.Repo
+  alias Dbservice.SessionSchedules
+  alias Dbservice.Sessions.SessionSchedule
+
+  action_fallback(DbserviceWeb.FallbackController)
+
+  use PhoenixSwagger
+
+  alias DbserviceWeb.SwaggerSchema.SessionSchedule, as: SwaggerSchemaSessionSchedule
+
+  def swagger_definitions do
+    Map.merge(
+      SwaggerSchemaSessionSchedule.session_schedule(),
+      SwaggerSchemaSessionSchedule.session_schedules()
+    )
+  end
+
+  swagger_path :index do
+    get("/api/session-schedule")
+
+    parameters do
+      params(:query, :string, "The id of a session",
+        required: false,
+        name: "id"
+      )
+    end
+
+    response(200, "OK", Schema.ref(:SessionSchedules))
+  end
+
+  def index(conn, params) do
+    today_day_of_week = Calendar.Date.day_of_week_name(Date.utc_today())
+
+    query =
+      from(m in SessionSchedule,
+        where: m.day_of_week == ^today_day_of_week,
+        order_by: [asc: m.id],
+        offset: ^params["offset"],
+        limit: ^params["limit"]
+      )
+
+    query =
+      Enum.reduce(params, query, fn {key, value}, acc ->
+        case String.to_existing_atom(key) do
+          :offset -> acc
+          :limit -> acc
+          atom -> from(u in acc, where: field(u, ^atom) == ^value)
+        end
+      end)
+
+    session_schedule = Repo.all(query)
+    render(conn, "index.json", session_schedule: session_schedule)
+  end
+
+  swagger_path :create do
+    post("/api/session-schedule")
+
+    parameters do
+      body(:body, Schema.ref(:SessionSchedule), "Session Schedule to create", required: true)
+    end
+
+    response(201, "Created", Schema.ref(:SessionSchedules))
+  end
+
+  def create(conn, params) do
+    with {:ok, %SessionSchedule{} = session_schedule} <-
+           SessionSchedules.create_session_schedule(params) do
+      conn
+      |> put_status(:created)
+      |> put_resp_header("location", Routes.session_schedule_path(conn, :show, session_schedule))
+      |> render("show.json", session_schedule: session_schedule)
+    end
+  end
+
+  swagger_path :show do
+    get("/api/session-schedule/{sessionScheduleId}")
+
+    parameters do
+      sessionScheduleId(:path, :integer, "The id of the session schedule record", required: true)
```

Comment:
We need one to be able to retrieve session schedule for a batch (most common use case)

Reply:
Already implemented. So there are 5 types of API endpoints:
1) Index API (GET /api/session-schedule):
- Purpose: Retrieve a list of session schedules
- Query parameters can be anything we want to filter (like `batch_id`, `session_id`, `start_time` etc. etc.)

2) Create API (POST /api/session-schedule):
- Purpose: Create a new session schedule.

3) Show API (GET /api/session-schedule/{sessionScheduleId}):
- Purpose: Retrieve details of a specific session schedule based on id (PK) of session schedule

4) Update API (PATCH /api/session-schedule/{sessionScheduleId}):
- Purpose: Update an existing session schedule.

5) Delete API (DELETE /api/session-schedule/{sessionScheduleId}):
- Purpose: Delete a session schedule.

This is the 3rd one listed above. If we want to filter session schedule based on batch, we will use the first one (Index API) by sending `batch_id` as query param.